src/lib/reducer.js

```diff
@@ -85,27 +105,31 @@ function collectAddSubItems(preOperator, node, collected, precision) {
   }
 }
 
-
+/**
+ * @param {import('../parser').CalcNode} node
+ * @param {number} precision
+ */
 function reduceAddSubExpression(node, precision) {
+  /** @type Collectible[] */
   const collected = [];
   collectAddSubItems('+', node, collected, precision);
 
-  const withoutZeroItem = collected.filter((item) => !(isValueType(item.node.type) && item.node.value === 0));
+  const withoutZeroItem = collected.filter((item) => !(isValueType(item.node) && item.node.value === 0));
   const firstNonZeroItem = withoutZeroItem[0]; // could be undefined
 
   // prevent producing "calc(-var(--a))" or "calc()"
   // which is invalid css
   if (!firstNonZeroItem ||
     firstNonZeroItem.preOperator === '-' &&
-    !isValueType(firstNonZeroItem.node.type)) {
+    !isValueType(firstNonZeroItem.node)) {
       const firstZeroItem = collected.find((item) =>
-        isValueType(item.node.type) && item.node.value === 0);
-      withoutZeroItem.unshift(firstZeroItem)
+        isValueType(item.node) && item.node.value === 0);
+      withoutZeroItem.unshift(/** @type Collectible*/(firstZeroItem))
```

Comment:
Potential bug? I don't see why `firstZeroItem` could not be `undefined`. Without this case TypeScript complains every time you get an element out of `collected`.

Reply:
hm, but `firstZeroItem` can be `undefined`, we should add check `firstNonZeroItem && firstNonZeroItem.preOperator === '-'`

Reply:
I've replaced the cast with checking if `firstZeroItem` is truthy before unshifting it. It makes no sense no add an `undefined` or `null` item to the array anyway.